pkg/controller/ec2/natgateway/controller.go

```diff
@@ -0,0 +1,204 @@
+	errStatusUpdate     = "cannot update status of the NATGateway resource"
+	errCreate           = "failed to create the NATGateway resource"
+	errDelete           = "failed to delete the NATGateway resource"
+	errUpdateTags       = "failed to update tags for the NATGateway resource"
+	errDeleteTags       = "failed to delete tags for NATGateway resource"
+)
+
+// SetupNatGateway adds a controller that reconciles NatGateways.
+func SetupNatGateway(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1beta1.NATGatewayGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1beta1.NATGateway{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1beta1.NATGatewayGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: ec2.NewNatGatewayClient}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(config aws.Config) ec2.NatGatewayClient
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1beta1.NATGateway)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awscommon.GetConfig(ctx, c.kube, mg, aws.StringValue(cr.Spec.ForProvider.Region))
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: c.newClientFn(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client ec2.NatGatewayClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mgd.(*v1beta1.NATGateway)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	response, err := e.client.DescribeNatGatewaysRequest(&awsec2.DescribeNatGatewaysInput{
+		NatGatewayIds: []string{meta.GetExternalName(cr)},
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ec2.IsNatGatewayNotFoundErr, err), errDescribe)
+	}
+
+	// in a successful response, there should be one and only one object
+	if len(response.NatGateways) != 1 {
+		return managed.ExternalObservation{}, errors.New(errNotSingleItem)
+	}
+
+	observed := response.NatGateways[0]
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errSpecUpdate)
+		}
+	}
```

Comment:
```suggestion
```

We do this check&update operation in case late-initialization changes anything. But it seems there is no need for late-init in this resource, so you can just skip doing that since that `if` will always return false.